src/System.Data.Common/tests/DbCommandTests.cs

```diff
@@ -0,0 +1,166 @@
+                }
+            }
+
+            public override CommandType CommandType
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override bool DesignTimeVisible
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override UpdateRowSource UpdatedRowSource
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            protected override DbConnection DbConnection
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            protected override DbParameterCollection DbParameterCollection
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            protected override DbTransaction DbTransaction
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override void Cancel()
+            {
+                throw new NotImplementedException();
+            }
+
+            public override int ExecuteNonQuery()
+            {
+                throw new NotImplementedException();
+            }
+
+            public override object ExecuteScalar()
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void Prepare()
+            {
+                throw new NotImplementedException();
+            }
+
+            protected override DbParameter CreateDbParameter()
+            {
+                throw new NotImplementedException();
+            }
+
+            protected override DbDataReader ExecuteDbDataReader(CommandBehavior behavior)
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        [Fact]
+        public void CanBeFinalized()
+        {
+            FinalizingCommand.CreateAndRelease();
+            GC.Collect();
+            GC.WaitForPendingFinalizers();
+            Assert.True(_wasFinalized);
```

Comment:
I assume but am not 100% sure that there are appropriate barriers in place such that a static written in a finalizer will have its state fully visible when WaitForPendingFinalizers completes.  I'd suggest marking _wasFinalized as volatile, if for no other reason than to communicate that it's being written by one thread and read by another.